app/src/main/java/com/example/concordia_campus_guide/Fragments/LocationFragment/LocationFragmentViewModel.java

```diff
@@ -270,14 +261,56 @@ public LatLng getSGWZoomLocation() {
     }
 
 
-    public void setListOfPOI(@PoiType String poiType) {
-        //TODO: Filter poi_list to only contain the 10 closest POIs
-        List<WalkingPoint> listOfPOI = appDatabase.walkingPointDao().getAllPointsForPointType(poiType);
-        this.poiList.postValue(listOfPOI);
+    public void setListOfPOI(@PoiType String poiType, Context context) {
+        List<WalkingPoint> allPOI = appDatabase.walkingPointDao().getAllPointsForPointType(poiType);
+        setCurrentPOIIcon(poiType, context);
+        this.poiList.postValue(getPOIinOrder(allPOI));
+    }
+
+    private PriorityQueue<WalkingPoint> getPOIinOrder(List<WalkingPoint> allPOI) {
+        PriorityQueue<WalkingPoint> orderedList = new PriorityQueue<>((WalkingPoint p1, WalkingPoint p2) -> {
+            Coordinates currentCoordinates = new Coordinates(currentLocation.getLatitude(), currentLocation.getLongitude());
+            double distanceFromO1 = p1.getCoordinate().getEuclideanDistanceFrom(currentCoordinates);
+            double distanceFromO2 = p2.getCoordinate().getEuclideanDistanceFrom(currentCoordinates);
+
+            //Compare walking points: If p1 is closer to the current location than p2, it will have a higher position in priority queue
+            if (distanceFromO1 < distanceFromO2) return -1;
+            else if (distanceFromO1 > distanceFromO2) return 1;
+            return 0;
+        });
+        orderedList.addAll(allPOI);
+        return orderedList;
     }
 
-    public LiveData<List<WalkingPoint>> getListOfPOI() {
+
+    LiveData<PriorityQueue<WalkingPoint>> getListOfPOI() {
         return poiList;
     }
 
+    private void setCurrentPOIIcon(@PoiType String poiType, Context context) {
+        currentPOIIcon = getCustomSizedIcon("point_of_interest_icons/poi_" + poiType.toLowerCase() + ".png", context, 60, 60);
+    }
+
+    BitmapDescriptor getCurrentPOIIcon() {
+        return currentPOIIcon;
+    }
+
+    void setCurrentLocation(Location currentLocation) {
+        this.currentLocation = currentLocation;
+    }
+
+    private BitmapDescriptor getCustomSizedIcon(String filename, Context context, int height, int width){
+        InputStream deckFile = null;
+        BitmapDescriptor smallMarkerIcon = null;
+        try {
+            deckFile = context.getAssets().open(filename);
+            Bitmap b = BitmapFactory.decodeStream(deckFile);
+            Bitmap smallMarker = Bitmap.createScaledBitmap(b, width, height, false);
+            smallMarkerIcon = BitmapDescriptorFactory.fromBitmap(smallMarker);
+            deckFile.close();
+        } catch (IOException e) {
+            LOGGER.log(Level.WARNING, e.getMessage(), e);
```

Comment:
we should be consistent in the logger that we use. For instance, in my DirectionsApiDataParser class, I use the Log object. Did we agree on one?

Reply:
I agree. We should come to a conclusion and then change our loggers to reflect that.

Reply:
Should we create an improvement task for it?